server/service.go

```diff
@@ -116,10 +117,41 @@ func (m *BoostService) handleRoot(w http.ResponseWriter, req *http.Request) {
 	fmt.Fprintf(w, `{}`)
 }
 
+// handleStatus sends calls to the status endpoint of every relay.
+// It returns OK if at least one returned OK, and returns KO otherwise.
 func (m *BoostService) handleStatus(w http.ResponseWriter, req *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusOK)
-	fmt.Fprintf(w, `{}`)
+	var wg sync.WaitGroup
+	var numSuccessRequestsToRelay uint32
+
+	// At the end, we wait for every routine and return status according to relay's ones.
+	defer func() {
```

Comment:
i'm not a big fan of this defer because of the indirection -- does it provide any advantage over putting the code just after the range loop? this would feel simpler to me intuitively 🤔

Reply:
I agree, let's not add spaghetti code while it's possible ^^